### PR TITLE
extended 3.9.1 definition

### PR DESCRIPTION
**Pull request by Haringat**
see ac94c07 commit message

### CODE REVIEW

File: 3.9.1/index.d.ts [hunk 1]
```diff
@@ -3,45 +3,11 @@
...
-   */
-  (): void;
-  /**
-   * Run a previously defined task
-   * @param taskName The name of the task to run (optional)
-   * @deprecated will be removed in gulp 4.0. Use task dependencies instead.
-   */
-  (taskName: string): void;
-}
+declare type task = string | ((outEvt: OutEvent) => void);
```
Comment (felixfbecker) [inline 3.9.1/index.d.ts]: Uppercase types, write out `outEvent`

File: 3.9.1/index.d.ts [hunk 2]
```diff
@@ -57,8 +23,23 @@ declare interface Gulp {
...
+   * @deprecated will be removed in gulp 4.0. Use task dependencies instead.
+   */
+  start: typeof Orchestrator.prototype.start;
+  /**
+   * watches all files in the specified glob(s) for changes and runs the given task(s).
+   * @param globs
+   * @param tasks
+   */
+  watch(globs?: string | Array<string>, tasks?: task | Array<task>): EventEmitter;
+  watch(globs?: string | Array<string>, opt?: WatchOptions, tasks?: task | Array<task>): EventEmitter;
```
Comment (felixfbecker) [inline 3.9.1/index.d.ts]: prefer shorter array type syntax `string[]`

Comment (felixfbecker) [reply 3.9.1/index.d.ts]: Pretty sure `watch` returns an instance of `Chokidar`

**Review by felixfbecker — CHANGES_REQUESTED** [review]

File: 3.9.1/index.d.ts (hunk 2 above)
Comment (blakeembrey) [reply 3.9.1/index.d.ts]: @felixfbecker Only the newest major version uses chokidar. Remember that `watch` is still using the old version of the dependency which uses `gaze`.

Comment (blakeembrey) [reply 3.9.1/index.d.ts]: Reference: https://github.com/gulpjs/vinyl-fs/blob/315e76a248e9d150f228e4ef50b61de15dc55123/package.json#L16 and https://github.com/gulpjs/glob-watcher/blob/195a14d51b734fd2ce762f44ec00de97ecb56da7/index.js#L5

Comment (Haringat) [reply 3.9.1/index.d.ts]: @felixfbecker When I followed up on the types for the return type I came to the conclusion that it is an EventEmitter.
And on the array notation topic: It doesn't make sense as I see it. Writing x[] looks like some abstract syntax-only thing to define an array (as you see in PHP or Java). But in ECMAScript that's not the case. Array is an actual class and by writing the Array&lt;x&gt; notation you express that it's an object of type Array that contains "somethings" (may be instances of objects or primitives) of type x. Beyond that it is often easier in more complex type notations to use generic notation because in short notation you might end up needing braces in the types which makes it harder to understand.
TL;DR: I don't really see the point in writing x[] rather than Array&lt;x&gt;. Can you give me objective reasons for that notation?

Comment (blakeembrey) [reply 3.9.1/index.d.ts]: @Haringat It's a bit clearer to use `string[]` over `Array<string>` (if only for conciseness). For more complex type shapes, I'd definitely recommend using the generic syntax. FWIW, the TypeScript type hints will always shorten it to `string[]` regardless I've found, so perhaps that's a reason?

Comment (Haringat) [reply 3.9.1/index.d.ts]: Not really, as different IDEs handle it differently. I know that VSCode behaves as you said and always uses bracket notation but IntelliJ IDEA/Webstorm uses whatever you wrote in the declaration.

Comment (felixfbecker) [reply 3.9.1/index.d.ts]: @Haringat The VS Code TypeScript IntelliSense is powered by the generic TypeScript language service, just like Atom, Sublime and many others. This is not IDE-specific, that is the official IntelliSense service by TypeScript that is bundled with the compiler on npm. Apparently Webstorm has built its own IntelliSense for TypeScript, but that is not a good reference.
It is simply shorter and also has been used for a long time in JSDoc long before TypeScript existed

Comment (felixfbecker) [reply 3.9.1/index.d.ts]: I personally don't care so much about what style is being used, but I care about consistency, so if a project uses a certain style this style should be used consistently

**Review by felixfbecker — APPROVED** [review]
